Products/ZenRRD/zencommand.py

```diff
@@ -657,20 +657,22 @@ def _parse_error(self, datasources, failure):
                 ",".join(ds.name for ds in datasources),
                 "timeout",
             )
-            results = []
-            for ds in datasources:
-                parsedResults = ParsedResults()
-                parsedResults.events.append(
-                    makeCmdTimeoutEvent(self._devId, ds),
-                )
-                results.append((ds, failure, parsedResults))
-            return results
+            results = [(ds, ParsedResults()) for ds in datasources]
+        else:
+            log.error(
+                "Command failed  device=%s datasource=%s error=%s reason=%s",
+                self._devId, datasources[0].name, failure.type, failure.value,
+            )
+            results = [(ds, ParsedResults()) for ds in datasources]
+            severity = Clear
```

Comment:
I have avoided adding new events.  If the released code didn't send an event for a particular error, these changes will continue that behavior.

The only behavior change I did make was the one requested in the ticket: timeout events for one datasource do not get cleared by the timeout clear events of another datasource.

Reply:
The code didn't send an event before because users had access to logs.  With cloud, we need to be *much* more proactive about sending events for things that impact users and that users might be able to do something about.  Even without that argument, our code is far from perfect.  Assuming it is doing (or not doing) something for a valid reason we can't see (or find after some reasonable investigation) as an excuse to not fix a valid issue we do see is a bad idea, IMHO.

Reply:
This sounds like a bigger issue than addressing ssh timeout issues.  It just not zencommand that would need these changes.

I think adding additional events to signal different collection error situations should probably go into a different ticket.